tests/e2e/specs/wcpay/shopper/shopper-checkout-free-cart.spec.js

```diff
@@ -0,0 +1,83 @@
+/**
+ * External dependencies
+ */
+import config from 'config';
+
+const {
+	shopper,
+	merchant,
+	clearAndFillInput,
+	uiUnblocked,
+} = require( '@woocommerce/e2e-utils' );
+
+/**
+ * Internal dependencies
+ */
+import {
+	setupCheckout,
+	fillCardDetails,
+	fillCardDetailsWCB,
+} from '../../../utils/payments';
+import { shopperWCP, merchantWCP, checkPageExists } from '../../../utils';
+
+const productName = config.get( 'products.simple.name' );
+const billingDetails = config.get( 'addresses.customer.billing' );
+const card = config.get( 'cards.basic' );
+
+describe( 'Checkout with free Cart & after Cart updates on Checkout page', () => {
+	beforeAll( async () => {
+		try {
+			await checkPageExists( 'checkout-wcb' );
+		} catch ( error ) {
+			await merchant.login();
+			await merchantWCP.addNewPageCheckoutWCB();
+			await merchant.logout();
+		}
+	} );
+
+	beforeEach( async () => {
+		await shopper.goToShop();
+		await shopper.addToCartFromShopPage( productName );
+		await shopper.goToCart();
+		await clearAndFillInput( '#coupon_code', 'free' );
+		await expect( page ).toClick( 'button[name="apply_coupon"]' );
+	} );
+
+	afterAll( async () => {
+		// Clear the cart at the end so it's ready for another test
+		await shopperWCP.emptyCart();
```

Comment:
We're adding to card before each test and emptying only after all are done - is that intended?

Reply:
Each test ends in Place Order, so it is not causing any issue if the tests are successful. Similar to the existing `tests/e2e/specs/wcpay/shopper/shopper-checkout-purchase.spec.js`. Since we are only checking "Order Received", having multiple items in the cart in case one of the test fails does not affect the next tests in both cases. Can still change it to be extra future proof, if required